Load device selection dynamically on Ports page

KayckMatias opened the pull request:
This pull request is to add the realtime search function for device filter on the ports page, the same function is present on other pages but not on this one.

After add:
![2e9133bd-66ca-4791-bc1e-11a85ec901fb](https://user-images.githubusercontent.com/48569093/190237766-3f59b536-6f51-4920-91fc-0703749d25b4.jpeg)

![2893360d-fe9e-4bca-be9b-af7efcab6f7f](https://user-images.githubusercontent.com/48569093/190237774-a5f29674-f685-4525-aefc-b8f57ccdff1e.jpeg)

DO NOT DELETE THE UNDERLYING TEXT

#### Please note

> Please read this information carefully. You can run `./lnms dev:check` to check your code before submitting.

- [x] Have you followed our [code guidelines?](https://docs.librenms.org/Developing/Code-Guidelines/)
- [x] If my Pull Request does some changes/fixes/enhancements in the WebUI, I have inserted a screenshot of it.
- [x] If my Pull Request makes discovery/polling/yaml changes, I have added/updated [test data](https://docs.librenms.org/Developing/os/Test-Units/).

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`
After you are done testing, you can remove the changes with `./scripts/github-remove`.  If there are schema changes, you can ask on discord how to revert.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/librenms/librenms?pullRequest=14353) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/librenms/librenms?pullRequest=14353) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/librenms/librenms?pullRequest=14353) it.</sub>

murrant reviewed (changes requested):
Nice improvement, but:
1. This leaves the old query(s) to fill the select box
2. This does not use the init_select2 helper
3. This does not use the newer select2 backend

KayckMatias commented:
> Nice improvement, but:
> 
> 1. This leaves the old query(s) to fill the select box
> 2. This does not use the init_select2 helper
> 3. This does not use the newer select2 backend (which is automatically used with init_select2)

So I should use init select2 to do this filtering and remove the old query, right?

KayckMatias commented:
> Nice improvement, but:
> 
> 1. This leaves the old query(s) to fill the select box
> 2. This does not use the init_select2 helper
> 3. This does not use the newer select2 backend (which is automatically used with init_select2)

I can use only select2 without ajax data? Since librenms itself already makes the query before loading the page? Or only init_select2?

KayckMatias commented:
can use only select2 without ajax data? Since librenms itself already makes the query before loading the page? Then I could remove the old query and use select2() ajax or only init_select2?

KayckMatias commented:
> Nice improvement, but:
> 
> 1. This leaves the old query(s) to fill the select box
> 2. This does not use the init_select2 helper
> 3. This does not use the newer select2 backend (which is automatically used with init_select2)

Updated to init_select2 and removed old query

murrant commented:
Nice work! I want this finished for the release today, so putting the last bit of polish on it.

Need to use "DeviceCache" to load devices (soon to be renamed to DeviceRepository)
Fix xss from injecting $vars['device_id'] into the page
Handle invalid device_id
Load select2 later, so target id exists
Load select2 on graphs page too
